gsitemap.php

```diff
@@ -1,907 +1,987 @@
+            if (!Configuration::deleteByName($key)) {
+                return false;
+            }
+        }
+
+        $hook = new Hook(Hook::getIdByName(self::HOOK_ADD_URLS));
+        if (Validate::isLoadedObject($hook)) {
+            $hook->delete();
+        }
+
+        return parent::uninstall() && $this->removeSitemap();
+    }
+
+    /**
+     * Delete all the generated Sitemap files  and drop the addon table.
+     * @return boolean
+     */
+    public function removeSitemap()
+    {
+        $links = Db::getInstance()->ExecuteS('SELECT * FROM `'._DB_PREFIX_.'gsitemap_sitemap`');
+        if ($links) {
+            foreach ($links as $link) {
+                if (!@unlink($this->normalizeDirectory(_PS_ROOT_DIR_).$link['link'])) {
+                    return false;
+                }
+            }
+        }
+        if (!Db::getInstance()->Execute('DROP TABLE `'._DB_PREFIX_.'gsitemap_sitemap`')) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function getContent()
+    {
+        /* Store the posted parameters and generate a new Google Sitemap files for the current Shop */
+        if (Tools::isSubmit('SubmitGsitemap')) {
+            Configuration::updateValue('GSITEMAP_FREQUENCY', pSQL(Tools::getValue('gsitemap_frequency')));
+            Configuration::updateValue('GSITEMAP_INDEX_CHECK', '');
+            Configuration::updateValue('GSITEMAP_CHECK_IMAGE_FILE', pSQL(Tools::getValue('gsitemap_check_image_file')));
+            $meta = '';
+            if (Tools::getValue('gsitemap_meta')) {
+                $meta .= implode(', ', Tools::getValue('gsitemap_meta'));
+            }
+            Configuration::updateValue('GSITEMAP_DISABLE_LINKS', $meta);
+            $this->emptySitemap();
+            $this->createSitemap();
+
+        /* if no posted form and the variable [continue] is found in the HTTP request variable keep creating sitemap */
+        } elseif (Tools::getValue('continue')) {
+            $this->createSitemap();
+        }
+
+        /* Empty the Shop domain cache */
+        if (method_exists('ShopUrl', 'resetMainDomainCache')) {
+            ShopUrl::resetMainDomainCache();
+        }
+
+        /* Get Meta pages and remove index page it's managed elsewhere (@see $this->_getHomeLink()) */
+        $store_metas = array_filter(Meta::getMetasByIdLang((int)$this->context->cookie->id_lang), function ($meta) {
+            return $meta['page'] != 'index' ;
+        });
+
+        $this->context->smarty->assign(
+            array(
+                'gsitemap_form' => './index.php?tab=AdminModules&configure=gsitemap&token='.Tools::getAdminTokenLite('AdminModules').'&tab_module='.$this->tab.'&module_name=gsitemap',
+                'gsitemap_cron' => 'http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true)._MODULE_DIR_.'gsitemap/gsitemap-cron.php?token='.Tools::substr(Tools::encrypt('gsitemap/cron'), 0, 10).'&id_shop='.$this->context->shop->id,
```

Comment:
You can replace this:
```php
'http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true)
```
with this:
```php
$this->context->link->getBaseLink()
```

Reply:
Then the output will become: <code> https://localhost/1741/test//1741/test/modules/gsitemap/gsitemap-cron.php?token=98344edcbb&id_shop=2</code>

what do I do wrong?

Reply:
Fixed

Reply:
@eternoendless can you please check this commit https://github.com/PrestaShop/gsitemap/pull/53/commits/6df978b8f042de9bc67b683a6b82ffa55379f8c0 ?